src/dusk/serde_support.rs

```diff
@@ -0,0 +1,123 @@
+extern crate alloc;
+
+use alloc::format;
+use alloc::string::String;
+
+use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{AffinePoint, ExtendedPoint, Fr};
+
+impl Serialize for AffinePoint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s = hex::encode(self.to_bytes());
+        serializer.serialize_str(&s)
+    }
+}
+
+impl<'de> Deserialize<'de> for AffinePoint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let s = String::deserialize(deserializer)?;
+        match hex::decode(s) {
+            Ok(bytes) => {
+                let bytes: [u8; 32] = bytes.try_into().map_err(|_| Error::custom(format!("Failed to deserialize AffinePoint: invalid byte length")))?;
+                AffinePoint::from_bytes(bytes)
+                    .into_option()
+                    .ok_or(Error::custom(
+                    "Failed to deserialize AffinePoint: invalid AffinePoint",
+                ))
+            }
+            Err(e) => Err(Error::custom(format!(
+                "Failed to deserialize AffinePoint: {e}"
+            ))),
+        }
+    }
+}
+
+impl Serialize for ExtendedPoint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        AffinePoint::from(self).serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for ExtendedPoint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        AffinePoint::deserialize(deserializer).map(|point| point.into())
```

Comment:
```suggestion
        AffinePoint::deserialize(deserializer).map(Into::into)
```

one small nit though, this can also be done and avoids creating an unnecessary closure.